### PR TITLE
Migrate to cloudtasks v2beta2

### LINKED ISSUE #1: move this functionality into TaskQueue (lease and patch)
https://github.com/talkiq/gcloud-rest/blob/5b2674895f5cf79c81e869612cd70ae72b76d7af/gcloud/rest/taskqueue/manager.py#L202

---

### PR DESCRIPTION
**Pull request by TheKevJames**
Fixes #1

### CODE REVIEW

File: gcloud/rest/taskqueue/manager.py [hunk 1]
```diff
@@ -104,13 +77,13 @@ def find_and_process_work(self):
                 tasks = [tasks[0]]
 
                 for task in to_burn:
-                    log.info('burning task %s', task.get('id'))
+                    log.info('burning task %s', task.get('name'))
```
Comment (jonathan-johnston) [inline gcloud/rest/taskqueue/manager.py]: I assume this means that the task metadata has changed in the update? If so, would this impact the clients of this lib (i.e. `gc-*` services) or does separation of concern cover this?

File: gcloud/rest/taskqueue/queue.py [hunk 2]
```diff
@@ -41,12 +41,128 @@ def headers(self):
...
+        url = '{}/tasks:lease'.format(self.api_root)
+
+        body = {
+            'maxTasks': min(num_tasks, 1000),
+            'leaseDuration': '{}s'.format(lease_duration),
+            'responseView': 'FULL',
+        }
+        if task_filter:
+            # TODO: this doesn't seem to work?
+            body['filter'] = task_filter
```
Comment (jonathan-johnston) [inline gcloud/rest/taskqueue/queue.py]: The task filter doesn't seem to be appended to the body, or the filter isn't applied properly downstream?

Comment (TheKevJames) [reply gcloud/rest/taskqueue/queue.py]: The filter is not applied. We don't need that capability for v1, so I'll likely leave this as-is.

File: gcloud/rest/taskqueue/manager.py (hunk 1 above)
Comment (TheKevJames) [reply gcloud/rest/taskqueue/manager.py]: Indeed. _Theoretically_, the point of this lib in the first place was to enforce the separation of concerns, but we'll have to double-check when we bump this dependency in the `gc-*` services.

**Comment by TheKevJames** [comment]
RFC (@jonathan-johnston ): The task [insert payload format](https://github.com/talkiq/gcloud-rest/pull/19/files#diff-377392a2059e91dec6e91f6a024b1189R106) is a bit stringent in order to support both Py2 and Py3 while conforming to the gcloud API. Currently, that function takes a payload of the form `my_task = {'apple': 'sauce', 'other_param': 42}; payload = json.dumps(my_task).encode()` ie. a bytes-encoded json-string. We could modify this to accept any format, do you have any suggestions? 

Note that we have no control over the _output_ of the task API, ie. when we get a task from the API we need to perform the [following operations](https://github.com/talkiq/gcloud-rest/pull/19/files#diff-302a39cd9b18431238a67bfa4dbfc25aR87) to get the json blob back, ie. b64decode, decode, jsonify.

I suppose the jsonification/un-jsonification is not a part of the scope of this library.. but what are your thoughts on handling these conversions? We should likely have the TaskQueue output the same format as in inputs, but how much of that work should we do vs. have consumers do?

**Comment by jonathan-johnston** [comment]
Personally, I'm a fan of the conversion to JSON because it means that the client has a consistent interface and not have to deal with encoding.

A question I have is this: Does the new API accept unicode payloads? In the Python2 case, doing a `decode` on the result of the base64 encode returns a unicode string.

**Comment by TheKevJames** [comment]
Yes, it accepts unicode and strings, but not bytes.

Ah -- the conversion to/from JSON should not be a part of the TaskQueue since theoretically there is no requirement that task payloads be jsonish. Its the conversion to/from base64 and bytes which I think we might(?) want to handle -- the task payload _must_ be a base64-encoded unicode/string, so that's the part we _might_ want to include in the TaskQueue.

The TaskManager, on the other hand, is effectively a consumer of the TaskQueue, so that's where the JSON encode/decode would live.

**Pull request merged**